### PR TITLE
PL-MASC ToHit mods and Movement mods match IntOps p.84

### LINKED ISSUE #2505: 47.14 Prosthetic leg MASC issues
Interstellar Operations page 83-84 states that "Attacks _against_ infantry units equipped with PL-MASC suffer a +1 to-hit modifier to account for these troopers' improved ability to dodge and weave in combat." Code for a +1 to-hit modifier related to PL-MASC [exists](https://github.com/MegaMek/megamek/blob/a034cc6a9ed6c905d7a305145841a8b341666bb5/megamek/src/megamek/common/actions/WeaponAttackAction.java#L3917), but gives the +1 to-hit malus to the infantry unit equipped with PL-MASC, rather than to units attacking said unit. 

PL-MASC is also very specifically limited to _foot_ infantry platoons (by both the text on IntOps p83 and [this rules confirmation thread](https://bg.battletech.com/forums/supplementary-rules/answered-io-prosthetic-leg-masc-rules-intent/)), and should not be valid for other motive types. Not certain how difficult it is to code such restrictions; this particular issue might also be more of a MegaMekLab issue (discovered both of these issues while creating a custom infantry unit with an unreasonable number of implants).

### PR DESCRIPTION
**Pull request by pakfront**
Fixes #2505

### CODE REVIEW

File: megamek/src/megamek/common/Infantry.java [hunk 1]
```diff
@@ -414,12 +414,14 @@ public int getWalkMP(boolean gravity, boolean ignoreheat, boolean ignoremodulara
                 && (getMovementMode() != EntityMovementMode.INF_JUMP)) {
             mp = Math.max(mp - 1, 0);
         }
+        //  PL-MASC IntOps p.84
         if ((null != getCrew())
                 && hasAbility(OptionsConstants.MD_PL_MASC)
-                && ((getMovementMode() == EntityMovementMode.INF_LEG)
-                    || (getMovementMode() == EntityMovementMode.INF_JUMP))) {
+                && getMovementMode() == EntityMovementMode.INF_LEG
+                    && isConventionalInfantry()) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/Infantry.java:421]:
```suggestion
                && getMovementMode().isLegInfantry()
                && isConventionalInfantry()) {
```

No need for the extra spacing now.

File: megamek/src/megamek/common/actions/WeaponAttackAction.java [hunk 2]
```diff
@@ -4288,10 +4281,18 @@ else if ((atype != null)
...
 
         // infantry squads are also hard to hit
-        if ((te != null) && te.isConventionalInfantry() && ((Infantry) te).isSquad()) {
+        if ((te instanceof Infantry) && te.isConventionalInfantry() && ((Infantry) te).isSquad()) {
             toHit.addModifier(1, Messages.getString("WeaponAttackAction.SquadTarget"));
         }
 
+        // pl-masc makes foot infantry harder to hit - IntOps p.84
+        if (te instanceof Infantry && te.hasAbility(OptionsConstants.MD_PL_MASC)
+                && te.getMovementMode() == EntityMovementMode.INF_LEG
```
Comment (Windchild292) [inline megamek/src/megamek/common/actions/WeaponAttackAction.java:4290]:
```suggestion
        if ((te instanceof Infantry) && te.hasAbility(OptionsConstants.MD_PL_MASC)
                && te.getMovementMode().isLegInfantry()
```

Our code style uses brackets when it comes to multiple operators.

**Review by Windchild292 — CHANGES_REQUESTED** [review]
Just a few things first

**Comment by pakfront** [comment]
Suggs accepted.

**Review by Windchild292 — APPROVED** [review]

**Pull request merged**